src/main/kotlin/no/nav/familie/ef/sak/api/beregning/VedtakDto.kt

```diff
@@ -1,53 +1,83 @@
-        val aktivitet: String,
-        val periodeType: String
-)
-
-
-fun List<Vedtaksperiode>.fraDomene(): List<VedtaksperiodeDto> =
-        this.map {
-            VedtaksperiodeDto(
-                    årMånedFra = YearMonth.from(it.datoFra),
-                    årMånedTil = YearMonth.from(it.datoTil),
-                    aktivitet = it.aktivitet,
-                    periodeType = it.periodeType,
-            )
-        }
+class Avslå(val resultatType: ResultatType = ResultatType.AVSLÅ,
+            val avslåBegrunnelse: String) : VedtakDto()
+
+fun VedtakDto.tilVedtak(behandlingId: UUID): Vedtak = when (this) {
+    is Avslå -> Vedtak(behandlingId = behandlingId,
+                       avslåBegrunnelse = this.avslåBegrunnelse,
+                       resultatType = ResultatType.AVSLÅ)
+    is Innvilget -> Vedtak(
+            behandlingId = behandlingId,
+            periodeBegrunnelse = this.periodeBegrunnelse,
+            inntektBegrunnelse = this.inntektBegrunnelse,
+            resultatType = ResultatType.INNVILGE,
+            perioder = PeriodeWrapper(perioder = this.perioder.tilDomene()),
+            inntekter = InntektWrapper(inntekter = this.inntekter.tilInntektsperioder()))
+    is Henlegge -> throw Feil("Kan ikke sette vedtak $this då det har feil type", "Kan ikke sette vedtak $this då det har feil type", HttpStatus.BAD_REQUEST)
+}
+
+fun Vedtak.tilVedtakDto(): VedtakDto =
+    when (this.resultatType) {
+        ResultatType.INNVILGE -> Innvilget(
+                resultatType = this.resultatType,
+                periodeBegrunnelse = this.periodeBegrunnelse ?: "",
+                inntektBegrunnelse = this.inntektBegrunnelse ?: "",
+                perioder = (this.perioder ?: PeriodeWrapper(emptyList())).perioder.fraDomene(),
+                inntekter = (this.inntekter ?: InntektWrapper(emptyList())).inntekter.tilInntekt())
+        ResultatType.AVSLÅ -> Avslå(
+                resultatType = this.resultatType,
+                avslåBegrunnelse = this.avslåBegrunnelse ?: ""
+        )
+        else -> throw Feil("Kan ikke sette vedtaksresultat som $this - ikke implementert")
+
+}
 
-fun List<VedtaksperiodeDto>.tilDomene(): List<Vedtaksperiode> =
-        this.map {
-            Vedtaksperiode(
-                    datoFra = it.årMånedFra.atDay(1),
-                    datoTil = it.årMånedTil.atEndOfMonth(),
-                    aktivitet = it.aktivitet,
-                    periodeType = it.periodeType,
-            )
+
+private class VedtakDtoDeserializer : StdDeserializer<VedtakDto>(VedtakDto::class.java) {
+
+    override fun deserialize(p: JsonParser, ctxt: DeserializationContext?): VedtakDto {
+        val mapper = p.codec as ObjectMapper
+        val node: JsonNode = mapper.readTree(p)
+        val resultatTypeNode = node.get("resultatType")
+        return when (mapper.readValue<ResultatType>(resultatTypeNode.toString())) {
+            ResultatType.INNVILGE -> mapper.readValue<Innvilget>(node.toString())
```

Comment:
Hær har man læst in det som ett træd som man gør till stræng och læser in det igen, det ær møjligt att gøra om det direkt till ett objekt med
```
ResultatType.INNVILGE -> mapper.treeToValue(node, Innvilget::class.java)
ResultatType.AVSLÅ -> mapper.treeToValue(node, Avslå::class.java)
```